tests/test_initialization.py

```diff
@@ -1330,3 +1333,177 @@ def test_warning_name_custom_model_with_custom_name(self, custom_module, recwarn
+        device_set_empty = {p.device.type for p in model.parameters()}
+        logits_empty = model(**inputs).logits
+
+        assert device_set_empty == device_set_not_empty
+        assert torch.allclose(logits_empty, logits_not_empty)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_load_adapter_empty_init_works(self, device, inputs, lora_path, lora_config):
+        model = self.get_model().to(device)
+        inputs = {k: v.to(device) for k, v in inputs.items()}
+
+        torch.manual_seed(0)
+        model = get_peft_model(model, lora_config)
+        model.load_adapter(lora_path, adapter_name="other", torch_device=device)
+        model.set_adapter("other")
+        model.eval()
+        device_set_not_empty = {p.device.type for p in model.parameters()}
+        logits_not_empty = model(**inputs).logits
+
+        del model
+
+        model = self.get_model().to(device)
+        torch.manual_seed(0)
+        model = get_peft_model(model, lora_config)
+        model.load_adapter(lora_path, adapter_name="other", init_empty=True, torch_device=device)
+        model.set_adapter("other")
+        model.eval()
+        device_set_empty = {p.device.type for p in model.parameters()}
+        logits_empty = model(**inputs).logits
+
+        assert device_set_empty == device_set_not_empty
+        assert torch.allclose(logits_empty, logits_not_empty)
+
+    @pytest.mark.parametrize("device", devices)
+    def test_inject_adapter_empty_init_works(self, device, inputs, lora_path, lora_config):
+        # external libs like transformers and diffusers use inject_adapter_in_model, let's check that this also works
+        model = self.get_model().to(device)
+        inputs = {k: v.to(device) for k, v in inputs.items()}
+
+        torch.manual_seed(0)
+        model = get_peft_model(model, lora_config)
+        model.load_adapter(lora_path, adapter_name="other", torch_device=device)
+        model.set_adapter("other")
+        model.eval()
+        device_set_not_empty = {p.device.type for p in model.parameters()}
+        logits_not_empty = model(**inputs).logits
+
+        del model
+
+        torch.manual_seed(0)
+        model = self.get_model().to(device)
+        inject_adapter_in_model(lora_config, model, init_empty=True)
+        device_set_before_loading = {p.device.type for p in model.parameters()}
+        # at this stage, lora weights are still on meta device
+        assert device_set_before_loading == {"meta", device}
+
+        state_dict = load_file(lora_path / "adapter_model.safetensors")
+        remapped_dict = {}
+        prefix = "base_model.model."
+        for key, val in state_dict.items():
+            new_key = key[len(prefix) :]
+            # module_name, _, param_name = new_key.rpartition(".")
+            # new_key = f"{module_name}.default.{param_name}"
+            remapped_dict[new_key] = val.to(device)
+        errors = set_peft_model_state_dict(model, remapped_dict, init_empty=True)
```

Comment:
Oh okay. So, `set_peft_model_state_dict()` has to be called with `init_empty=True` if the injection (`inject_adapter_in_model()`) was made with `init_empty=True`. I think that should be okay.